### ISSUE TITLE
(🐞) `PLW1641` not reported if method defined within `if`

### ISSUE THREAD

**Issue by KotlinIsland** [astral-sh/ruff#14882]
```py
class A:
    if True:

        def __eq__(self, other): ...
```
this doesn't report any error
